File: Ejercicio_0012/Polish_alphabet.java
package Ejercicio_0012;

import javax.swing.JOptionPane;

public class Polish_alphabet {
    public static void main(String[] args) {
        String cadena = JOptionPane.showInputDialog("Escribe la cadena:");

        String[] polish = {"ą", "ć", "ę", "ł", "ń", "ó", "ś", "ź"};
        String[] replacements = {"a", "c", "e", "l", "n", "o", "s", "z"};

        String new_cadena = cadena;

        for (int i = 0; i < polish.length; i++) {
            new_cadena = new_cadena.replace(polish[i], replacements[i]);
        }

        JOptionPane.showMessageDialog(null, "Resultado: \n\n" + new_cadena);
    }
}

File: Ejercicio_0012/Polish_alphabet.js
const texto = prompt("Escribe la cadena:");

const polish = {
    "ą" : "a",
    "ć" : "c",
    "ę" : "e",
    "ł" : "l",
    "ń" : "n",
    "ó" : "o",
    "ś" : "s",
    "ź" : "z"
}

let new_texto = "";

let l = 1;

for (let i = 0; i < texto.length; i++) {

    let caracter = texto[i];

    if (polish[caracter]) {
        new_texto = new_texto + polish[caracter];
    } else {
        new_texto = new_texto + caracter;
    }
}

alert("Resultado:\n\n" + new_texto);